### PR TITLE
Add Circle `move()`/`move_ip()`

### PR DESCRIPTION
**Pull request by itzpr3d4t0r**
This PR adds the move/move_ip functions to the Circle object. It also adds the possibility of doing equality/inequality checks and an internal function for creating circle objects starting from three values.

## Credits

**Geometry Project:**
For code, docs and tests:
@novialriptide @Emc2356 @itzpr3d4t0r @ScriptLineStudios @avaxar @Matiiss @newpaxonian @maqa41 @blankRiot96
Also thanks to @Starbuck5 for kickstarting the idea and the occasional help!

**Functionality added in this PR**
- `circle.c`
https://github.com/pygame-community/pygame-geometry/blame/main/src_c/circle.c
Credits to @Emc2356 @itzpr3d4t0r @novialriptide
- `geometry.pyi`
https://github.com/pygame-community/pygame-geometry/blame/main/geometry.pyi
Credits to @Emc2356 @itzpr3d4t0r @novialriptide @ScriptLineStudios @avaxar 
- `geometry_test.py`
https://github.com/pygame-community/pygame-geometry/blame/main/test/test_circle.py
Credits to @itzpr3d4t0r @novialriptide
- `geometry.rst`
https://github.com/pygame-community/pygame-geometry/blame/main/docs/circle.rst
Credits to @itzpr3d4t0r

### CODE REVIEW

**Review by MyreMylar — APPROVED** [review]
LGTM 👍

File: src_c/circle.c [hunk 1]
```diff
@@ -489,6 +535,26 @@ pg_circle_setdiameter(pgCircleObject *self, PyObject *value, void *closure)
...
 
+static PyObject *
+pg_circle_richcompare(PyObject *self, PyObject *other, int op)
+{
+    pgCircleBase c1, c2;
+    if (!pgCircle_FromObject(self, &c1) || !pgCircle_FromObject(other, &c2)) {
+        Py_RETURN_NOTIMPLEMENTED;
+    }
+
+    int equal = c1.x == c2.x && c1.y == c2.y && c1.r == c2.r;
```
Comment (ankith26) [inline src_c/circle.c:546]: All these attributes are floats (correct me if I'm wrong here) and if they are, doing this sort of comparison could be a bad idea? Maybe we should do a difference-under-epsilon kinda comparison instead?

Comment (itzpr3d4t0r) [reply src_c/circle.c]: Thanks, I've added a function to account for that.

**Review by ankith26 — COMMENTED** [review]
looks like this PR has conflicts now

File: src_c/circle.c [hunk 2]
```diff
@@ -548,6 +594,35 @@ pg_circle_setdiameter(pgCircleObject *self, PyObject *value, void *closure)
...
+    const double e = 1e-6;
+    return fabs(a - b) < e || fabs(a - b) <= e * MAX(fabs(a), fabs(b));
+}
+
+static PyObject *
+pg_circle_richcompare(PyObject *self, PyObject *other, int op)
+{
+    pgCircleBase c1, c2;
+    if (!pgCircle_FromObject(self, &c1) || !pgCircle_FromObject(other, &c2)) {
+        Py_RETURN_NOTIMPLEMENTED;
```
Comment (ankith26) [inline src_c/circle.c:610]: The general python convention for implementing comparison of non-matching types is:
`==` should be `False`, `!=` should be true and all other operations should raise the not implemented error

I know pygame doesn't stick to this convention in a lot of places, but we can certainly consider this for new code

**Review by ankith26 — APPROVED** [review]
Left another nitpick about richcompare, though I'm happy with `move[_ip]` so I will go ahead and approve this, thanks for the PR :tada:

**Pull request merged**